src/hotspot/share/memory/archiveBuilder.cpp

```diff
@@ -613,9 +712,124 @@ void ArchiveBuilder::make_klasses_shareable() {
 
       if (log_is_enabled(Debug, cds, class)) {
         ResourceMark rm;
-        log_debug(cds, class)("klasses[%4d] = " PTR_FORMAT " %s", i, p2i(to_target(ik)), ik->external_name());
+        log_debug(cds, class)("klasses[%4d] = " PTR_FORMAT " %s", i, p2i(to_requested(ik)), ik->external_name());
+      }
+    }
+  }
+}
+
+uintx ArchiveBuilder::buffer_to_offset(address p) const {
+  address requested_p = to_requested(p);
+  assert(requested_p >= _requested_static_archive_bottom, "must be");
+  return requested_p - _requested_static_archive_bottom;
+}
+
+uintx ArchiveBuilder::any_to_offset(address p) const {
+  if (is_in_mapped_static_archive(p)) {
+    return p - _mapped_static_archive_bottom;
+  }
+  return buffer_to_offset(p);
+}
```

Comment:
Is it possible to add an `assert(DynamicDumpSharedSpaces)` after the `if` statement at line 728?

Reply:
Added per suggestion.